modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Twin/src/ModuleProcess.cs

```diff
@@ -89,15 +90,17 @@ public class ModuleProcess : IProcessControl {
                     _reset = new TaskCompletionSource<bool>();
                     var module = hostScope.Resolve<IModuleHost>();
                     var logger = hostScope.Resolve<ILogger>();
-                    var config = new Config(_config);
+                    var moduleConfig = hostScope.Resolve<IModuleConfig>();
+                    var identity = hostScope.Resolve<IIdentity>();
                     logger.Information("Initiating prometheus at port {0}/metrics", kTwinPrometheusPort);
                     var server = new MetricServer(port: kTwinPrometheusPort);
                     try {
-                        server.StartWhenEnabled(config, logger);
+                        server.StartWhenEnabled(moduleConfig, logger);
                         // Start module
                         var product = "OpcTwin_" +
                             GetType().Assembly.GetReleaseVersion().ToString();
-                        kTwinModuleStart.Inc();
+                        kTwinModuleStart.WithLabels(identity.SiteId ?? "",
```

Comment:
We should add this into the ModuleHost class.   I am not sure if these are even set at this point as I believe Start has to be called first.

Reply:
well, DeviceId and ModuleId are set (now, I did some changes to make it happen), the SiteId is set to the default value, but this might be overwritten with the value from the twin's definition. the ModuleHost has a different metric that is kind-of similar

Reply:
I would prefer it to be in the ModuleHost start/stop methods as that is where these values are set.  And yes, site is dynamic and can change  To make sure we have correct numbers I would not add it as label, only deviceid, gw, moduleid which are fixed over the course of the module host running.

Reply:
ModuleHost has a separate metrics on start/stop methods. I removed the SiteId now from all metrics.